portfolio/src/main/java/com/google/sps/data/Query.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.data;
+
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.Entity;
+import java.io.IOException;
+
+import java.util.stream.Collectors;
+import java.util.List;
+import java.util.Comparator;
+
+public class Query {
+
+    private String text;
+
+    public Query (String text) {
+        this.text = text;
+    }
+
+    public static Query makeQuery(String captions, int size) {
+        LanguageServiceClient languageService;
+        try {
+            languageService = LanguageServiceClient.create();
+        } catch (IOException e) {
+            return null;
+        }
+
+        Document doc =
+            Document.newBuilder().setContent(captions).setType(Document.Type.PLAIN_TEXT).build();
+        AnalyzeEntitiesRequest request = AnalyzeEntitiesRequest.newBuilder()
+            .setDocument(doc)
+            .build();
+        AnalyzeEntitiesResponse response = languageService.analyzeEntities(request);
```

Comment:
Thought: it looks like this is where the NLP does its magic. Maybe add a comment describing its costs, in terms of units, latency, etc.?

Reply:
Thanks for the suggestion! Added.